packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -37,6 +37,8 @@ interface ISecondaryYAxisValues {
   secondaryYScaleOptions?: { yMinValue?: number; yMaxValue?: number };
 }
 
+const SUPPORTED_PLOT_TYPES = ['pie', 'bar', 'scatter', 'heatmap', 'sankey', 'indicator', 'histogram'];
```

Comment:
>SUPPORTED_PLOT_TYPES [](http://example.com/codeflow?start=6&length=20)

this looks duplicate of switch case logic.

Reply:
The idea behind this change is that Plotly renders a line when the data type is invalid (e.g., 'line' or undefined) and the data can be represented as a line. The isLineData function helps identify data that can be represented as a line, but I thought there could be other data types with the same structure that we might handle differently. This is why I checked whether the data type is not one of the supported plot types. That said, I'm still not sure if this reasoning is entirely correct.